gw_eccentricity/eccDefinition.py

```diff
@@ -1285,33 +1442,217 @@ def get_good_extrema(self, pericenters, apocenters,
             pericenters, apocenters)
         return pericenters, apocenters
 
-    def get_interp(self, oldX, oldY, allowExtrapolation=False,
-                   interpolator="spline"):
-        """Get interpolant.
-
-        A wrapper of utils.get_interpolant with check_kwargs=False.
-        This is to make sure that the checking of kwargs is not performed
-        everytime the interpolation function is called. Instead, the kwargs
-        are checked once in the init and passed to the interpolation
-        function without repeating checks.
+    def get_rational_fit_for_extrema(self, x, y, data_name=None):
+        """Get rational fit with adaptive numerator and denominator degree.
+
+        This function ensures that the rational fit uses the optimal degree for
+        the numerator and denominator. It first checks for nonmonotonicity in
+        the fit's derivative and lowers the degrees if necessary. Only if the
+        fit for the initial degree was monotonic (no reduction of degree
+        needed), it attempts to increase the degrees to find a higher-degree
+        fit. If increasing the degrees causes nonmonotonicity, it reverts to
+        the previous valid monotonic fit.
```

Comment:
A bit more explanation would be nice. Maybe describe the form of the rational polinomial and say whether you increase/decrease the degree of denominator and numerator at the same time. Also mention what the minimum and  maximum degrees are.

Reply:
Done